src/ParametersHandler/YarpImplementation/src/YarpImplementation.cpp

```diff
@@ -15,6 +17,73 @@
 
 using namespace BipedalLocomotion::ParametersHandler;
 
+bool YarpImplementation::stringToDuration(const std::string& string, std::chrono::nanoseconds& time)
+{
+    using namespace std::chrono_literals;
+
+    constexpr auto logPrefix = "[YarpImplementation::stringToDuration]";
+    // \\. means that \ is used as escape character for the regexp and again as escape character for
+    // the string literal.
+    const std::regex expression("[0-9]{2}:[0-9]{2}:[0-9]{2}\\.[0-9]{6}\\b");
```

Comment:
I would suggest adding an explanation of what this expression is expected to match. This might help the future you reading again the code.

Reply:
Also, I guess we do not allow setting a ``nanosecond`` from configuration file, right?

Reply:
I added the explanation of the regex in https://github.com/ami-iit/bipedal-locomotion-framework/pull/630/commits/3a7b778950664255e59f8fe8028ffd1a2c39d21c